src/object.d

```diff
@@ -3241,6 +3241,269 @@ template RTInfo(T)
     enum RTInfo = null;
 }
 
+private template Floating1(T)
```

Comment:
I see @WalterBright replied to that providing a rationale. At any rate, something we can do right now:

```d
    // inside _adCmpT
    static int compare(X f1, X f2)
    {
        static if (is(T == cfloat) || is(T == cdouble) || is(T == creal))
        {
            ...
        }
        else static if (is(T == float) || is(T == double) || is(T == real))
        {
            ...
        }
        else static assert(false, "Internal error");
    }
```

Then use compare.